src/test/java/uk/gov/justice/digital/delius/integration/secure/OffendersResource_getOffenderNsisByCrn.java

```diff
@@ -0,0 +1,113 @@
+@ActiveProfiles("dev-seed")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class OffendersResource_getOffenderNsisByCrn {
+
+    private static final String OFFENDERS_PATH = "/offenders/crn/%s/convictions/%s/nsis/%s";
+
+    private static final Long KNOWN_CONVICTION_ID = 2500295345L;
+
+    private static final String KNOWN_OFFENDER = "X320741";
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Value("${test.token.good}")
+    private String validOauthToken;
+
+    @Before
+    public void setup() {
+        RestAssured.port = port;
+        RestAssured.basePath = "/secure";
+        RestAssured.config = RestAssuredConfig.config().objectMapperConfig(
+                new ObjectMapperConfig().jackson2ObjectMapperFactory((aClass, s) -> objectMapper));
+    }
+
+    @Test
+    public void getGetOffenderNsisByCrnAndConvictionId() {
+        final var nsiWrapper = given()
+                .auth()
+                .oauth2(validOauthToken)
+                .contentType(APPLICATION_JSON_VALUE)
+                .when()
+                .get(String.format(OFFENDERS_PATH, KNOWN_OFFENDER, KNOWN_CONVICTION_ID, "BRES,BRE"))
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .extract()
+                .body()
+                .as(NsiWrapper.class);
+
+        assertThat(nsiWrapper.getNsis()).hasSize(2);
+    }
+
+    @Test
+    public void getGetOffenderNsisByCrnAndConvictionIdLimitCode() {
+        final var nsiWrapper = given()
+            .auth()
+            .oauth2(validOauthToken)
+            .contentType(APPLICATION_JSON_VALUE)
+            .when()
+            .get(String.format(OFFENDERS_PATH, KNOWN_OFFENDER, KNOWN_CONVICTION_ID, "BRES"))
+            .then()
+            .statusCode(HttpStatus.OK.value())
+            .extract()
+            .body()
+            .as(NsiWrapper.class);
+
+        assertThat(nsiWrapper.getNsis()).hasSize(1);
+    }
+
+    @Test
+    public void getGetOffenderNsisByCrnAndUnknownConvictionId_ReturnsEmptyCollection() {
+        final var nsiWrapper = given()
+            .auth()
+            .oauth2(validOauthToken)
+            .contentType(APPLICATION_JSON_VALUE)
+            .when()
+            .get(String.format(OFFENDERS_PATH, KNOWN_OFFENDER, "0", "BRES,BRE"))
+            .then()
+            .statusCode(HttpStatus.OK.value())
+            .extract()
+            .body()
+            .as(NsiWrapper.class);
+
+        assertThat(nsiWrapper.getNsis()).hasSize(0);
+    }
+
+    @Test
+    public void getOffenderNsisByCrn_offenderNotFound_returnsNotFound() {
+        given()
+            .auth()
+            .oauth2(validOauthToken)
+            .contentType(APPLICATION_JSON_VALUE)
+            .when()
+            .get(String.format(OFFENDERS_PATH, "X777777", KNOWN_CONVICTION_ID, "BRES,BRE"))
+            .then()
+            .statusCode(HttpStatus.NOT_FOUND.value());
+    }
+}
```

Comment:
Could we add a test case for a known offender and conviction with no breach NSIs. i.e. returns an empty list

Reply:
Will do.